charts/powerdns/templates/deployment.yaml

```diff
@@ -0,0 +1,143 @@
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ template "powerdns.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ tpl .Values.image.tag . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          volumeMounts:
+          - name: config
+            mountPath: /etc/pdns/conf.d
+          env:
+{{- if .Values.powerdns.postgres.enabled }}
+            - name: PGSQL_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: postgres_host
+            - name: PGSQL_PORT
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: postgres_port
+            - name: PGSQL_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: postgres_username
+            - name: PGSQL_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: postgres_password
+            - name: PGSQL_DATABASE
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: postgres_database
+            - name: PGSQL_DNSSEC
+              value: {{if (.Values.powerdns.dnssec)}}"yes"{{else}}"no"{{ end }}
+{{ end }}
+{{- if .Values.powerdns.mysql.enabled }}
+            - name: MYSQL_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: mysql_host
+            - name: MYSQL_USER
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: mysql_username
+            - name: MYSQL_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: mysql_password
+            - name: MYSQL_DB
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "powerdns.fullname" . }}
+                  key: mysql_database
+            - name: MYSQL_DNSSEC
+              value: {{if (.Values.powerdns.dnssec)}}"yes"{{else}}"no"{{ end }}
+{{ end }}
+          ports:
+            - name: dns-tcp
+              containerPort: 53
+              protocol: TCP
+            - name: dns-udp
+              containerPort: 53
+              protocol: UDP
+{{ if (.Values.powerdns.api_key) }}
```

Comment:
It appears that if the `powerdns.api_key` value is not set, we will never have a `dns-webserver` port defined and thus the probes won't properly work.

If this is required, we may not be able to do automated testing of the chart unless we have some default value for `powerdns.api_key` or work probes in a way that don't require this port?

What do you think?

Reply:
Hmm. Theoretically you don't need the API to run powerdns, but I think we should be opinionated here and always run it. What if I auto generate the key if it's not provided?

Thoughts?